### PR TITLE
test/plugin autotest 

### PR DESCRIPTION
**Pull request by dmsolr**
Please answer these questions before submitting pull request

- Why submit this pull request?
- [ ] Bug fix
- [x] New feature provided
- [ ] Improve performance

- Related issues
#2905 
___
### New feature or improvement
- Describe the details and related test reports.

This PR works base on #2905 which @ascrutae does. The main purpose is to move the plugin-autotest back main repository.

1. There is a httpclient-scenario only here.
2. I don't add this to Jenkins's stages yet.
3. After this, I will add more scenarios for the tests.

### CODE REVIEW

File: Jenkinsfile-Agent-Test [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+        stage('Compile plugin-test tools Codes') {
+            steps {
+                sh './mvnw -f test/plugin/pom.xml -Pagent clean package -DskipTests'
+            }
+        }
+
+        stage('Run httpclient-scneario Tests') {
+            steps {
+                sh './test/plugin/run.sh httpclient-4.3.x-scneario'
```
Comment (kezhenxu94) [inline Jenkinsfile-Agent-Test]:
```suggestion
                sh 'bash test/plugin/run.sh httpclient-4.3.x-scneario'
```

**Comment by wu-sheng** [comment]
The first case passed. Congrats, good start. You could remove `WIP` of title, and let's polish this, and make it as an official step.

**Comment by wu-sheng** [comment]
This PR status looks good to me. @kezhenxu94 and @ascrutae please do final check, then we will run this again with right label, then prepare to merge.

**Comment by wu-sheng** [comment]
@kezhenxu94 After you approve, please open the jenkins test to run automatically.

**Comment by wu-sheng** [comment]
Waiting for another approve.

**Comment by wu-sheng** [comment]
We should start an issue to track existing plugin test cases immigration.

**Comment by dmsolr** [comment]
> We should start an issue to track existing plugin test cases immigration.

After this PR, we are going to immigrate `gson-plugin`, `jdk-plugin` or both.

**Comment by wu-sheng** [comment]
Submit two issues, let's tracke them. Then mysql with compose supportedv

**Comment by dmsolr** [comment]
I will open an issue to track every plugin immigrated status later and open PR for `compose`.

**Pull request merged**